src/server/game/Entities/Item/ItemTemplate.cpp

```diff
@@ -23,6 +23,15 @@
 
 #include "Packets/QueryPackets.h"
 
+bool ItemTemplate::HasSignature() const
+{
+    return GetMaxStackSize() == 1 &&
+        Class != ITEM_CLASS_CONSUMABLE &&
+        Class != ITEM_CLASS_QUEST &&
+        (Flags & ITEM_FLAG_NO_CREATOR) == 0 &&
+        ItemId != 6948 /*Hearthstone;*/
```

Comment:
Why the hearthstone? It doesn't have any signature.

Reply:
It's going to get a signature unless that line is there because of https://db.rising-gods.de/?spell=3286

Reply:
But the spell is casted by the npc on the player, in this case the name of the npc appears in the signature?

Reply:
https://github.com/TrinityCore/TrinityCore/commit/fdd8d2f3cc7108043f6bad036b994770f904866f#diff-6e2ad27531ee8eb7bafce8a41909816aR3072
originated here, HS is signed with the player name.

Reply:
![WoWScrnShot_031919_145154](https://user-images.githubusercontent.com/1158801/54611141-b375de80-4a56-11e9-802d-aba91ccefaac.jpg)

Reply:
That's with IdemId != 6948 commented out.